src/controllers/reportArticle.js

```diff
@@ -0,0 +1,55 @@
+import models from '../db/models';
+import errorResponse from '../helpers';
+
+const { Report } = models;
+
+/**
+ * @description A controller class for handling comment business logic
+ *
+ * @class Comment
+ */
+class ReportArticle {
+  /**
+   * @description - reports an article
+   *
+   * @static
+   * @param {object} request - request object
+   * @param {object} response - response object
+   *
+   * @returns {object} - error or success response
+   *
+   * @memberof Comment
+   */
+  static async report(request, response) {
+    let { type: reportType = '', context } = request.body;
+    reportType = reportType ? reportType.toLowerCase() : '';
+    context = context || '';
+
+    const reportTypes = ['spam', 'harrassment', 'rules violation', 'terrorism'];
```

Comment:
Hi @codeshifu, I love how you have been able to implement this feature, and I have also learned from your style of destructuring, but I think we could make the report types more flexible. Think of it this way, as a user, I should be able to report for whatever type of reasons. I would like to know what you think.

Reply:
I don't think so @frostyblok . Every app has terms and conditions the user is expected to agree and keep which rules violation covers for. Users should not be given the flexibility things that way else they might be abusing it.

Reply:
@okoroemeka Yes of course, as a developer, but the users are the one using the application. So I think it would be wise for user's satisfaction to be factored in here. We could have an "other" section, for the user to be able to make their complaints more specific. What do you think about that?

Reply:
Good job, @codeshifu !